PWGHF/Utils/utilsAnalysis.h

```diff
@@ -109,62 +109,52 @@ bool isSelectedTrackTpcQuality(T const& track, const int tpcNClustersFoundMin, c
+};
 
-  /// Mass selection of 2 or 3 prong canidates in triggered data analysis
-  /// \tparam nProngs switch between 2-prong and 3-prong selection
-  /// \param invMass is the invariant mass of the candidate
-  /// \param pdgMass is the pdg Mass of the candidate particle
-  /// \param pt is the pT of the candidate
-  /// \return true if candidate passes selection
-  template <uint8_t nProngs>
-  bool isCandidateInMassRange(const float& invMass, const double& pdgMass, const float& pt)
-  {
-    float ptMassCutMax{0.};
-    float ptDeltaMassMax{0.};
-    float nSigmaMax{0.};
-    o2::framework::LabeledArray<float> deltaMassPars;
-    o2::framework::LabeledArray<float> sigmaPars;
-
-    if constexpr (nProngs == 2) {
-      deltaMassPars = deltaMassPars2Prong;
-      sigmaPars = sigmaPars2Prong;
-      ptDeltaMassMax = ptDeltaMass2ProngMax;
-      ptMassCutMax = ptMassCut2ProngMax;
-      nSigmaMax = nSigma2ProngMax;
-    } else if constexpr (nProngs == 3) {
-      deltaMassPars = deltaMassPars3Prong;
-      sigmaPars = sigmaPars3Prong;
-      ptDeltaMassMax = ptDeltaMass3ProngMax;
-      ptMassCutMax = ptMassCut3ProngMax;
-      nSigmaMax = nSigma3ProngMax;
-    } else {
-      LOGF(fatal, "nProngs %d not supported!", nProngs);
-    }
-
-    float peakMean = (pt < ptDeltaMassMax) ? ((pdgMass + deltaMassPars.get("constant")) + deltaMassPars.get("linear") * pt) : pdgMass;
-    float peakWidth = sigmaPars.get("constant") + sigmaPars.get("linear") * pt;
+/// Configurable group to apply trigger specific cuts for 3-prong HF analysis
+struct HfTrigger3ProngCuts : o2::framework::ConfigurableGroup {
+  std::string prefix = "hfTrigger3ProngCuts"; // JSON group name
 
-    return (!(std::abs(invMass - peakMean) > nSigmaMax * peakWidth && pt < ptMassCutMax));
-  }
+  static constexpr float defaultDeltaMassPars3Prong[1][2] = {{-0.0025f, 0.0001f}};
+  static constexpr float defaultSigmaPars3Prong[1][2] = {{0.00796f, 0.00176f}};
+  o2::framework::Configurable<float> nSigma3ProngMax{"nSigma3ProngMax", 2, "Maximum number of sigmas for pT-differential mass cut for 3-prong candidates"};
+  o2::framework::Configurable<float> ptDeltaMass3ProngMax{"ptDeltaMass3ProngMax", 10., "Max pT to apply delta mass shift to PDG mass value for 3-prong candidates"};
+  o2::framework::Configurable<float> ptMassCut3ProngMax{"ptMassCut3ProngMax", 9999., "Max pT to apply pT-differential cut for 3-prong candidates"};
+  o2::framework::Configurable<o2::framework::LabeledArray<float>> deltaMassPars3Prong{"deltaMassPars3Prong", {defaultDeltaMassPars3Prong[0], 2, {"constant", "linear"}}, "delta mass parameters for HF 3-prong trigger mass cut"};
+  o2::framework::Configurable<o2::framework::LabeledArray<float>> sigmaPars3Prong{"sigmaPars3Prong", {defaultSigmaPars3Prong[0], 2, {"constant", "linear"}}, "sigma parameters for HF 3-prong trigger mass cut"};
```

Comment:
You can consider defining `HfTrigger2ProngCuts` and `HfTrigger3ProngCuts` with the same data members, except for the prefix. Then you can just pass the reference to the `struct` as argument of `isCandidateInMassRange` and get the configurables inside.

Reply:
Good suggestion! Implemented 👍